src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -184,7 +194,7 @@ DistributedTableSize(Oid relationId, char *sizeQuery)
 		totalRelationSize += relationSizeOnNode;
 	}
 
-	heap_close(relation, AccessShareLock);
+	heap_close(relation, NoLock);
```

Comment:
I don't think we particularly care whether the relation gets dropped immediately after calling this function. `pg_table_size` releases the lock.